File: app/channels/conversation_channel.rb
class ConversationChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    stream_from "conversations-#{current_user.id}"
    puts "def sub"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
    stop_all_streams
    puts "def unsub"
  end

  def speak(data)
    message_params = data['message'].each_with_object({}) do |element, hash|
      hash[element.values.first] = element.values.last
    end

    puts "hi i am def speak"
    puts message_params
    # ActionCable.server.broadcast(
    #                              "conversations-#{current_user.id}",
    #                              {
    #                                title: "from conversation channel",
    #                                message: message_params
    #                              })
    # Message.create(message_params)

  end
end

File: app/javascript/channels/conversation_channel.js
import consumer from "channels/consumer"
const conversationChannel = consumer.subscriptions.create("ConversationChannel", {
        connected() {
                // Called when the subscription is ready for use on the server
                console.log("connected to conversation channel");
        },

        disconnected() {
            // Called when the subscription has been terminated by the server
        },

        received(data) {
                // Called when there's incoming data on the websocket for this channel
                console.log(data['message']);
                submit_messages(data);
        },

        speak: function (message) {
            console.log('speak');
            console.log(message);
            return this.perform('speak', {
                message: message
            });

        }
})
window.conversationChannel = conversationChannel;

let submit_messages;
submit_messages = function (data) {
    var conversation = $('#conversations-list').find("[data-conversation-id='" + data['conversation_id'] + "']");
    console.log('conversation')
    conversation.find('.messages-list').find('ul').append(data['message']);
    console.log("logging the appender");
    conversation.find('textarea').val('');
    var messages_list = conversation.find('.messages-list');
    var height = messages_list[0].scrollHeight;
    messages_list.scrollTop(height);
}




